src/kibana/plugins/discover/components/field_chooser/field_chooser.js

```diff
@@ -21,10 +21,16 @@ define(function (require) {
         data: '=',
         state: '=',
         indexPattern: '=',
+        indexPatternList: '=',
         updateFilterInQuery: '=filter'
       },
       template: html,
-      controller: function ($scope) {
+      controller: function ($scope, $route) {
+        $scope.setIndexPattern = function (indexPattern) {
+          $scope.state.index = indexPattern;
```

Comment:
This sort of relies on `$scope.state` being initialized in a parent scope, which, in practice, should always be the case. However, I sort of don't like relying on an outside source. I could just as easily inject `AppState`, and initialize `$scope.state = new AppState();` at the top of the controller function. Would that be a better alternative?